lib/posix/mqueue.c

```diff
@@ -341,6 +347,82 @@ int mq_setattr(mqd_t mqdes, const struct mq_attr *mqstat,
 	return 0;
 }
 
+/**
+ * @brief Notify process that a message is available.
+ *
+ * See IEEE 1003.1
+ */
+int mq_notify(mqd_t mqdes, const struct sigevent *notification)
+{
+	int ret, detachstate;
+	mqueue_desc *mqd = (mqueue_desc *)mqdes;
+
+	if (mqd == NULL) {
+		errno = EBADF;
+		return -1;
+	}
+
+	mqueue_object *msg_queue = mqd->mqueue;
+
+	if (notification == NULL) {
+		if ((msg_queue->not.sigev_notify & SIGEV_MASK) == 0) {
+			errno = EINVAL;
+			return -1;
+		}
+		remove_notification(msg_queue);
+		return 0;
+	}
+
+	if ((msg_queue->not.sigev_notify & SIGEV_MASK) != 0) {
+		errno = EBUSY;
+		return -1;
+	}
+	if (notification->sigev_notify == SIGEV_SIGNAL) {
+		errno = ENOSYS;
+		return -1;
+	}
+	if (notification->sigev_notify_attributes != NULL) {
+		ret = pthread_attr_getdetachstate(notification->sigev_notify_attributes,
+						  &detachstate);
+		if (ret != 0) {
+			errno = ret;
+			return -1;
+		}
+		if (detachstate != PTHREAD_CREATE_DETACHED) {
```

Comment:
No need to check what the attr detachstate is before setting it
```suggestion
		ret = pthread_attr_getdetachstate(notification->sigev_notify_attributes,
						  &detachstate);
		if (ret != 0) {
			errno = ret;
			return -1;
		}
		if (detachstate != PTHREAD_CREATE_DETACHED) {
```

Reply:
That's the way `posix/timer.c` do it so maybe it should be changed there as well?